### PR TITLE
Move distribution build infra to CDK deployment

### PR DESCRIPTION
**Pull request by peternied**
### Description
Switching to CDK to support the additional aws tools needed to make build artifacts public
 
### Issues Resolved
* #453 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: bundle-workflow/infra/lib/distribution-build-stack.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+export interface DistributionBuildStackProps extends StackProps {
+  name: string,
+  bucketName?: string,
+  buildRoleName?: string,
+  bundleRoleName?: string,
+  testRoleName?: string,
+}
+
+export class DistributionBuildStack extends Stack {
```
Comment (peternied) [inline bundle-workflow/infra/lib/distribution-build-stack.ts:18]: If you are looking for the 'meat' of this change as it were - this is it.  Everything else is support for the CDK build/deployment process.

File: bundle-workflow/infra/lib/distribution-build-stack.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  ArnPrincipal, CanonicalUserPrincipal, IRole, PolicyStatement, Role,
+} from '@aws-cdk/aws-iam';
+import { Bucket } from '@aws-cdk/aws-s3';
+import {
+  Arn, CfnOutput, CfnParameter, Construct, Stack, StackProps,
+} from '@aws-cdk/core';
+
+export interface DistributionBuildStackProps extends StackProps {
+  name: string,
+  bucketName?: string,
```
Comment (peternied) [inline bundle-workflow/infra/lib/distribution-build-stack.ts:12]: These optional parameters are in place so we can deploy in an account that already has used the cloud formation template that I removed

File: bundle-workflow/infra/bin/distribution-build.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// arn:aws:iam::336984078605:role/JenkinsStackDev-MainNodeInstanceRole6FA3C5C9-1PQFWOWDYN42U
+
+const stacks: DistributionBuildStackProps[] = [{
+  name: 'DistributionBuildStack-Staging',
+  bucketName: 'teststatus-buildbucket-1otmnpvrkhfji',
+  buildRoleName: 'opensearch-build',
+  bundleRoleName: 'opensearch-bundle',
+  testRoleName: 'opensearch-test',
+  env: {
+    account: '791023489099',
```
Comment (peternied) [inline bundle-workflow/infra/bin/distribution-build.ts:15]: 99% sure we can include the account numbers in PR, but I need to confirm this.

File: bundle-workflow/infra/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Bundle Workflow Infrastructure
```
Comment (peternied) [inline bundle-workflow/infra/README.md:1]: Needs a real description of the design of this system

File: bundle-workflow/infra/test/distribution-build-stack.test.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { expect as expectCDK, matchTemplate, MatchStyle } from '@aws-cdk/assert';
+import * as cdk from '@aws-cdk/core';
+import { DistributionBuildStack } from '../lib/distribution-build-stack';
+
+test('Empty Stack', () => {
```
Comment (peternied) [inline bundle-workflow/infra/test/distribution-build-stack.test.ts:5]: Needs actual tests

**Review by peternied — COMMENTED** [review]
Here are some of the areas keeping this PR in draft.

**Comment by dblock** [comment]
Given #448 do you want to think about putting it at the root?

File: bundle-workflow/infra/bin/distribution-build.ts (hunk 3 above)
Comment (peternied) [reply bundle-workflow/infra/bin/distribution-build.ts]: FYI - We should not, using our internal repository to store values - converting these over to parameters

**Comment by peternied** [comment]
> Given #448 do you want to think about putting it at the root?

Moved to the root and ready for review!

**Review by dblock — DISMISSED** [review]

**Comment by peterzhuamazon** [comment]
What is the reason to have CDK here if we already have Jenkins.

**Comment by peternied** [comment]
> What is the reason to have CDK here if we already have Jenkins.

These changes are specific to the build process, built off of the existing cloud formation template that was side-by-side with these changes.  This location will likely change when the new repo opens, but this is codifying the changes we need for the 1.1.0 release.

**Comment by peterzhuamazon** [comment]
> existing cloud formation template that was side-by-side with these changes

Can you point me to those templates? In my understanding the bundle workflow is only running python and shell either locally or on CI.

**Comment by peternied** [comment]
@peterzhuamazon The CFN template is in this PR, it is the [deleted](https://github.com/opensearch-project/opensearch-build/pull/496/files#diff-76b1f952750e04f87f6f17e543ce3770f37e92432568c440753d19f123821d82).

**Comment by peternied** [comment]
@dblock @gaiksaya @peterzhuamazon Could you take a look?

**Review by dblock — APPROVED** [review]
I am not super versed in CDK so no detailed comments, YOLO.

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**